pkg/controllers/dlb/controller.go

```diff
@@ -73,10 +76,11 @@ func (c *controller) GetTotalObjectCount(ctx context.Context, clnt client.Client
 
 func (c *controller) NewDaemonSet(rawObj client.Object) *apps.DaemonSet {
 	devicePlugin := rawObj.(*devicepluginv1.DlbDevicePlugin)
+	nodeSelector := controllers.AddArchConstraintToNodeSelector(devicePlugin.Spec.NodeSelector)
```

Comment:
would it make sense to add NodeSelector to the plugin yaml and forget about setting it in the code?

Reply:
Aha, yes, it makes sense.